PSI Food waste/Pages/Forms/EditProduct.cshtml.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using PSI_Food_waste.Models;
+
+namespace PSI_Food_waste.Pages.Forms
+{
+    public class EditProductModel : PageModel
+    {
+        private readonly PSI_Food_waste.Data.ProductContext _context;
+
+        public EditProductModel(PSI_Food_waste.Data.ProductContext context)
+        {
+            _context = context;
+        }
+
+        [BindProperty]
+        public Product Product { get; set; }
+
+        public async Task<IActionResult> OnGetAsync(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            Product = await _context.Products.FirstOrDefaultAsync(m => m.PrID == id);
+
+            if (Product == null)
+            {
+                return NotFound();
+            }
+            return Page();
+        }
+
+        // To protect from overposting attacks, enable the specific properties you want to bind to.
+        // For more details, see https://aka.ms/RazorPagesCRUD.
+        public async Task<IActionResult> OnPostAsync()
+        {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
+            Product.DiscountedPrice = Product.Price * Product.Discount / 100;
+            _context.Attach(Product).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!ProductExists(Product.PrID))
+                {
+                    return NotFound();
+                }
+                else
+                {
+                    throw;
+                }
+            }
+
+            return RedirectToPage("./RestaurantVerified");
```

Comment:
can you help me fix this line? If I update the product it goes back to RestaurantVerified but on return currentRestaurant is null